cupy/cuda/cub.pyx

```diff
@@ -43,7 +43,19 @@ cdef extern from 'cupy_cub.h':
 ###############################################################################
 
 
+# TODO(leofang): remove this hack when CUB supports complex numbers
 def reduce_sum(core.ndarray x, out=None):
+    if x.dtype in (numpy.complex64, numpy.complex128):
```

Comment:
not too important, but `x.dtype.kind == 'c'` is a bit more efficient (~50 ns vs 160 ns on my system, at least in Python. I haven't tried timing it for Cython)

Reply:
Interesting. Thanks, @grlee77! I will find some time to time it in Cython. Didn't know this could make a difference.

But, more generally, this issue is related to #2463. So far CuPy has no coding guideline for a consistent choice. @toslunar, please let me know if you have any preference.

Reply:
I prefer `x.dtype.kind == 'c'` (if it is fast enough).

Reply:
This decision is no longer needed with the new PR #2538.